etna/core/middleware.py

```diff
@@ -122,3 +124,27 @@ def process_template_response(
             ),
         )
         return response
+
+
+class SearchMiddleware:
+    """
+    Handles the session settings for "back to search results" functionality in the record details page.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+
+        return response
+
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        if view_func.__name__ == record_detail_view.__name__:
+            # retain session var regardless of is exitence
+            return None
+
+        # delete session var a non-search view or function
+        if request.session.get("back_to_search_url", ""):
+            del request.session["back_to_search_url"]
```

Comment:
Do we need to do this? If the user searches again, the value will be updated. If they end up on a record detail page some other way, is remembering their search a big problem?

Reply:
Good catch @ababic Fixed to retain the session for all views.